File: wg_sharedavatar_cake_expt/MacCore/AvatarCore/bin/pyTests/data_receiver.py
# This Python file uses the following encoding: utf-8
from CorePython.CoreAPI import CoreAPI
from CorePython.CoreData import CoreData

import time

core = CoreAPI()
data_wrapper = None


def on_data_frame_arrived(resource, channel, frame):
    print(resource + ':' + channel)
    print(frame)


def stop():
    global data_wrapper
    data_wrapper.destroy()
    data_wrapper = None
    global core
    core.destroy_api()
    core = None
    print('Core destroy')


if __name__ == "__main__":
    core.init_api(app_path='../',port=0)
    data_wrapper = CoreData('local-data',
                            'app',
                            on_data_frame_arrived)
    data_wrapper.start()

    try:
        while True:
            time.sleep(1)
            pass
    except KeyboardInterrupt:  # Captures Ctrl + C
        stop()

File: wg_sharedavatar_cake_expt/MacCore/AvatarCore/bin/jsTests/data_receiver.js
/*
Data receiver
This example shows how to:
- Initialize avatarCore SDK
- Create a core data wrapper object for resource [test-resource]
- Start receiving data via a callback
- Destroy the SDK when pressing ctrl+c

Author: yamen@avatarin.com
*/

const CoreSDK = require("../CoreJS/CoreAPI");
const { CoreDataWrapper } = require("../CoreJS/CoreDataWrapper");

//setup input library
const readline = require('readline');
readline.emitKeypressEvents(process.stdin);
process.stdin.setRawMode(true);


//Initialize Core
CoreSDK.InitAPI(master = false, appPath = "../")

//when a data frame arrived, it will call this function
function cb_onFrameArrived(resource, channel, frame) {
    console.log("Received data frame: ", resource, ":", channel, " --> ", frame)

}

//Create a data wrapper object, this will handle data resources
let dataWrapper = new CoreDataWrapper("local-data", "app", cb_onFrameArrived);


//Start reading the data resource
dataWrapper.Start();


const ExitApp = () => {
    //destroy and clean up resources
    dataWrapper.Stop();
    dataWrapper.Destroy();
    dataWrapper = undefined
    //destroy and clean up the core
    CoreSDK.DestroyAPI()
    console.log("Core Destroyed!")
    process.exit(0)
};


process.stdin.on('keypress', (str, key) => {
    if (key.ctrl && key.name === 'c') {
        ExitApp()
    }
});

